<a href="https://colab.research.google.com/github/jennifer-algabre/flyrank-ml-internship/blob/main/work/notebooks/w01_research_question.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# ML-02 — Research Question and Provisional Lane

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

I chose the **Content Refresh Prioritization** lane. This lane focuses on helping identify which content pages may benefit most from review based on observable search performance signals. I chose it because prioritizing content updates is a practical decision that can support SEO and content teams. Rather than relying on a single metric, I want to explore how multiple signals can provide better decision support.

In [ ]:
# No code required for this section.

## 2. The question: decision, action, cost of a wrong call

The decision is which content pages should be prioritized for review or refresh.

The people who act on this decision are content editors or SEO teams. They can use the results to decide which pages to review first.

A wrong recommendation could cause important pages needing updates to be overlooked or lead to time being spent refreshing pages that do not need immediate attention.

Machine learning may help because search performance depends on many interacting signals that are difficult to capture with a single fixed rule.

In [ ]:
# No code required for this section.

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [ ]:
!git clone https://github.com/jennifer-algabre/flyrank-ml-internship.git
%cd flyrank-ml-internship

Cloning into 'flyrank-ml-internship'...
remote: Enumerating objects: 95, done.
remote: Counting objects: 100% (95/95), done.
remote: Compressing objects: 100% (66/66), done.
remote: Total 95 (delta 17), reused 80 (delta 13), pack-reused 0 (from 0)
Receiving objects: 100% (95/95), 1.84 MiB | 2.86 MiB/s, done.
Resolving deltas: 100% (17/17), done.
/content/flyrank-ml-internship/flyrank-ml-internship


In [ ]:
!find . -name "*.csv"

./outputs/refresh_queue_sample.csv
./data/raw/content_refresh_anonymized.csv


In [ ]:
import pandas as pd

# Load the starter dataset
df = pd.read_csv("data/raw/content_refresh_anonymized.csv")

# Basic dataset information
print(f"Total pages: {len(df):,}")
print(f"Total clients: {df['client_id'].nunique()}")

# Distribution of trend directions
print("\nTrend direction distribution:")
print(df["trend_direction"].value_counts())

# Average CTR
print(f"\nAverage CTR: {df['ctr'].mean():.2f}%")

Total pages: 30,000
Total clients: 32

Trend direction distribution:
trend_direction
down      16262
stable     5962
up         4388
new        2236
flat       1152
Name: count, dtype: int64

Average CTR: 0.51%


The starter dataset contains **30,000 content pages** from **32 pseudonymized clients**.

The dataset includes pages with different performance trends, with **16,262 pages** showing a downward trend. Together with an average **CTR of 0.51%**, these observations suggest meaningful variation in content performance.

These observed statistics support my choice of the **Content Refresh Prioritization** lane because they indicate that multiple search performance signals may help prioritize which pages should be reviewed first.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

This project will provide **observed** and **measured** insights from the available search performance data. Any recommendations will be **directional** and intended to support decision-making about which content pages may be prioritized for review.

This project will **not** prove that updating a page will improve its search performance, determine how Google's ranking algorithm works, or make causal claims. The results are intended as **decision-support**, not as guarantees of future outcomes.

In [ ]:
# No code required for this section.